app/js/playlist/controllers/PlaylistCtrl.js

```diff
@@ -100,7 +132,7 @@ angular.module("FM.playlist.PlaylistCtrl", [
          * @method onEnd
          */
         $scope.onEnd = function onEnd(event, data) {
-            if ($scope.playlist[0].track.uri === data.uri) { // jshint ignore:line
+            if (($scope.playlist[0].track.uri === data.uri) && ($scope.page.pages === $scope.page.total)) { // jshint ignore:line
                 $scope.meta.total--;
                 $scope.meta.play_time = $scope.meta.play_time - $scope.playlist[0].track.duration; // jshint ignore:line
```

Comment:
As queue meta is not paginated, this should always be updated onEnd, only splicing the track from the list needs to be wrapped in this pagination condition.